### DISCUSSION TITLE
pk increase in every conflict on upsert

### DISCUSSION THREAD

**Discussion by canali83** [hasura/graphql-engine#8448] [Question]
Hello

I have a normal pk id as sequantial increase. when i use upsert with constraint on conflict update_columns:[] i see anormal growth on id column number. As i understood even there is a conflict postgres db auto create new id even not being used. 

How can i use upsert with sequential id increase?

Thanks

**Comment by canali83** [comment]
any suggestion?

**Comment by seanonthenet** [comment]
Are you saying that on an upsert like this:

```graphql
mutation upsert_article {
  insert_article (
    objects: [
      {
        title: "Article 1",
        content: "Updated article 1 content",
        published_on:"2018-10-12"
      }
    ],
    on_conflict:{
      constraint: article_title_key,
      update_columns: [content]
    }
  ) {
    returning {
      id
      title
      content
      published_on
    }
  }
}
```

...where there **IS** a conflict, your `id` **IS** being sequentially updated?

Because default functionality is that when an update happens on conflict the `id` should not be changed.

**Reply by canali83** [reply]
i use to insert multiple objects with update_columns: [] . if there is conflict i do nothing but in every conflict id is still increasing(in the background i think) that causing anormal growth instead of sequential growth like 1 5 11 21 etc...

**Reply by seanonthenet** [reply]
Ok. I get you. I guess Hasura is _trying_ the insert first, failing on the constraint, which increments the `id` in the background, and then switching to update of the row where the constraint matches. 

I agree it's not ideal. I doubt it would change any time soon. 

Is it critical that your app has ids that are sequentially updated by 1?

**Reply by canali83** [reply]
Yes, its critical because i make a check in every hour.
now i have 620773 rows but id is already reached to 17415491 and i don't know what is the limit of id field. i can do manual query check then do insert but its not an efficient way

**Reply by canali83** [reply]
any update for upsert to avoid pk increase on constraint situtation?
